File: masterclass/Demomvp/app/src/main/java/com/example/demo_mvp/home/HomeContract.kt
package com.example.demo_mvp.home

/**
 * MVP contract: defines View and Presenter through interfaces so neither depends
 * on the other's concrete type, and the Model never knows either exists.
 * See references/ui-pattern-mvp.md in the android-development skill.
 */
interface HomeContract {

    interface View {
        fun showLoading()
        fun hideLoading()
        fun showData(data: HomeUiData)
        fun showError(message: String)
    }

    interface Presenter {
        fun attachView(view: View)
        fun detachView()
        fun loadData()
        fun onTransactionClicked(transactionId: String)
    }
}

File: masterclass/Demo-mvp-iOS/Demo-mvp-iOS/Home/HomeContract.swift
import Foundation

@MainActor
protocol HomeViewProtocol: AnyObject {
    func showLoading()
    func hideLoading()
    func showData(_ data: HomeUiData)
    func showError(_ message: String)
}

@MainActor
protocol HomePresenterProtocol: AnyObject {
    func attachView(_ view: HomeViewProtocol)
    func detachView()
    func loadData()
    func onTransactionClicked(_ transactionId: String)
}
